core/include/traccc/clusterization/detail/sparse_ccl.hpp

```diff
@@ -56,7 +60,8 @@ inline unsigned int make_union(std::vector<unsigned int>& L, unsigned int e1,
 /// @param b the second cell
 ///
 /// @return boolan to indicate 8-cell connectivity
-inline bool is_adjacent(cell a, cell b) {
+template <typename cell_t>
+TRACCC_HOST_DEVICE inline bool is_adjacent(cell_t a, cell_t b) {
```

Comment:
:thinking: Why did you need to template the cell type? Don't we have just a single cell type in both host and device code? :confused:

I also wonder if passing the cells to this function with const references may be a better idea. Cells are small, but are they small enough for passing around by value? (I'm honestly not sure...)

Reply:
I templated them because we're passing just `cell` in the CPU code but `const cell` in the device. I can change it to receive `cell_t&` then? :thinking:

Reply:
As we discussed before lunch, let's just go back to passing `cell` objects by value to these functions. You just need to add `TRACCC_HOST_DEVICE` on them in this PR.